.github/workflows/ci.yml

```
@@ -218,3 +218,60 @@ jobs:
           --verbose --numprocesses auto --showlocals
         env:
           TEMP: "R:\\Temp"
+
+  tests-windows-non-utc:
+    name: tests / ${{ matrix.python }} / ${{ matrix.os }} / ${{ matrix.timezone }}
+    runs-on: ${{ matrix.os }}-latest
+
+    needs: [pre-commit, packaging, determine-changes]
+    if: >-
+      needs.determine-changes.outputs.tests == 'true' ||
+      github.event_name != 'pull_request'
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [Windows]
+        python:
+          # Commented out, since tests already run in UTC
+          # - 3.7
+          # - 3.8
+          # - 3.9
+          - "3.10"
+        timezone:
+          - "Singapore Standard Time"
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - uses: szenius/set-timezone@v1.0
```

```suggestion
      - uses: pradyunsg/set-timezone@v1.0
```

I've forked the package and allowed that fork, to reduce the potential number of users that could hijack what code is run in our CI pipeline.

The current list of allowed actions is:

actions/checkout@v2,
actions/setup-python@v2,
dessant/lock-threads@v3,
dorny/paths-filter@v2,
pradyunsg/auto-label-merge-conflicts@v3,
pradyunsg/set-timezone@v1.0,
pre-commit/action@v2.0.0,

Both paths-filter and lock-threads have decently large userbases, so I'm fine with the risk we take on via those.